### ISSUE TITLE
ConditionalField does not hide layer and tag attributes

### ISSUE THREAD

**Issue by VirsabiMunk** [Deadcows/MyBox#57]
As the title says - would be nice implementation :)

**Comment by Deadcows** [comment]
Well, such issue were opened once, but I considered this behavior is "intended".
Now I think that such thing would be certainly nice to have 👌🏻 so thanks you for bringing this up. 
I'll implement this in a few days I believe, I don't have much time at the moment

**Comment by VirsabiMunk** [comment]
Cool thanks! Will make it easy to make components that can check layers/tags or not. :)

**Comment by Deadcows** [comment]
Oh dang, I completely misunderstood what you requested for some reason (probably I should sleep a bit more 😅). I though you wanted to hide Separator or Header along with the property.

Ok then, I was unable to reproduce your issue. 
Can you share your code? 

        public bool A;
        [ConditionalField("A"), Tag] public string TA;
        [Tag] public string TB;

![](https://habrastorage.org/webt/bd/6e/lc/bd6elcmsmu7vbnvfb8ddops4xrw.gif)

**Comment by VirsabiMunk** [comment]
Apparently, tag needs to be the last attribute assigned. In this code, the conditionalfield for Tag works, but not for Layer;

public bool checkTag;
    [ConditionalField(nameof(checkTag)), Tag]
    public string CollideWithTag = "Untagged";

    public bool checkLayer;

    [Layer, ConditionalField(nameof(checkLayer))]
    public int CollideWithLayer;

**Comment by Deadcows** [comment]
For now I figured out that ConditionalField should always be the first attribute. I experimented with order, like this:
        `public bool B;
        [Tag(order = 1), ConditionalField(nameof(B), order = 2)] public string BB;`
but strangely with such setup fieldInfo of PropertyDrawer becomes null.
I'll leave this issue to figure out null problem later, for now I'll suggest to simply keep ConditionalField the first attribute.

Hopefully this will work in your case 🙄
